build/index.js

```diff
@@ -397,6 +378,14 @@ async function buildDocument(document, documentOptions = {}) {
 
   doc.modified = metadata.modified || null;
 
+  const root = document.fileInfo.root;
+  doc.source = {
+    hash: metadata.hash || null,
+    folder: document.fileInfo.folder,
+    repository_url: REPOSITORY_URLS[root],
+    branch: getCurrentGitBranch(root),
+  };
```

Comment:
I think you should keep the `injectSource` function and put all of these lines into that function. 
The reason is that the function you're in the middle of is is already long enough. So there's a pattern here of calling out to individual small functions. 
Yes, it's a bit crazy that we have functions that are only every used from one place. But I think it's a worthwhile thing to do to keep the main function (`buildDocument()`) reasonable to read through when you're debugging it.

Reply:
Done!